src/index.js

```diff
@@ -1,274 +0,0 @@
-			return res.send(filteredMensages.slice(-limit));
-		}
-
-		res.send(filteredMensages);
-	} catch (err) {
-		console.error(err);
-		res.sendStatus(500);
-	}
-});
-
-app.delete("/messages/:ID_DA_MENSAGEM", async (req, res) => {
-	try {
-		const id = req.params.ID_DA_MENSAGEM;
-		const user = stripHtml(req.headers.user).result;
-		const message = await db
-			.collection("messages")
-			.findOne({ _id: new ObjectId(id) });
-
-		if (message.from != user) {
-			return res.sendStatus(401);
-		}
-
-		await db.collection("messages").deleteOne({ _id: new ObjectId(id) });
-		res.sendStatus(200);
-	} catch (error) {
-		res.sendStatus(404);
-	}
-});
-
-app.put("/messages/:ID_DA_MENSAGEM", async (req, res) => {
-	try {
-		const id = req.params.ID_DA_MENSAGEM;
-		const from = stripHtml(req.headers.user).result;
-		const to = stripHtml(req.body.to).result;
-		const text = stripHtml(req.body.text).result;
-		const type = stripHtml(req.body.type).result;
-		const validFrom = await db
-			.collection("participants")
-			.findOne({ name: from });
-		const newMessage = {
-			from: from.trim(),
-			to,
-			text,
-			type,
-		};
-		const message = await db
-			.collection("messages")
-			.findOne({ _id: ObjectId(id) });
-
-		const validation = messageSchema.validate(newMessage, {
-			abortEarly: true,
-		});
-
-		if (validation.error) {
-			console.log(validation.error.message);
-			return res.sendStatus(422);
-		}
-
-		if (!validFrom) {
-			return res.sendStatus(422);
-		}
-
-		if (message.from != from) {
-			return res.sendStatus(401);
-		}
-
-		await db.collection("messages").updateOne(
-			{
-				to: message.to,
-				text: message.text,
-				type: message.type,
-			},
-			{ $set: newMessage },
-		);
-
-		res.sendStatus(200);
-	} catch (error) {
-		res.sendStatus(404);
-	}
-});
-
-app.post("/status", async (req, res) => {
-	try {
-		const user = stripHtml(req.headers.user).result;
-		const newStatus = { lastStatus: Date.now() };
-		const validUser = await db
-			.collection("participants")
-			.findOne({ name: user });
-
-		if (!validUser) {
-			return res.sendStatus(404);
-		}
-
-		const selectedUser = await db
-			.collection("participants")
-			.findOne({ name: user });
-
-		await db.collection("participants").updateOne(
-			{
-				lastStatus: selectedUser.lastStatus,
-			},
-			{ $set: newStatus },
-		);
-
-		res.sendStatus(200);
-	} catch (err) {
-		console.error(err);
-		res.sendStatus(500);
-	}
-});
-
-setInterval(deleteInativatedUsers, 15000);
-
-async function deleteInativatedUsers() {
-	try {
-		const participants = await db.collection("participants").find().toArray();
-		participants
-			.filter((el) => Date.now() - el.lastStatus > 10000)
-			.forEach((el) => {
-				const newStatus = {
-					from: el.name,
-					to: "Todos",
-					text: "sai da sala...",
-					type: "status",
-					time: dayjs().format("HH:mm:ss"),
-				};
-
-				db.collection("participants").deleteOne({ _id: ObjectId(el._id) });
-				db.collection("messages").insertOne(newStatus);
```

Comment:
Take care with this kind of code, it works here because the order between the asynchronous operations don't matter. However, lets say you wanted to be querying things from a database inside a loop and the exact order mattered. Loops don't work well with async operations, because these operations are not run inside javascript. So in a scenario like this, you would need to await for all promises to be finished before proceding and code would look something like this:

```javascript
const newUsers = await Promise.all(participants.map(async (participant) => {
  const newParticipant = await db.collection("participants").insertOne(newUser);
  return newParticipant;
}));
```